File: Homeworks/C#1/Operators-And-Expressions/PointInCircleOutRectangle/PointInCircleOutRectangle.cs
//Write an expression that checks for given point (x, y) 
//if it is within the circle K({1, 1}, 1.5) and out of the rectangle R(top=1, left=-1, width=6, height=2).

namespace PointInCircleOutRectangle
{
    using System;
    using System.Linq;
    class PointInCircleOutRectangle
    {
        const double radius = 1.5;
        const double xY = 1;
        static void Main(string[] args)
        {
            Console.Write("Enter the coordinates X and Y separated by a space. ");
            double[] coords = Console.ReadLine()
                .Trim()
                .Split(new char[] { ' ', '\t' }, StringSplitOptions.RemoveEmptyEntries)
                .Select(x => Double.Parse(x))
                .ToArray();
            Console.WriteLine(IsInsideCircle(coords[0], coords[1]) && IsOutsideRectangle(coords[0], coords[1]));
        }

        static bool IsInsideCircle(double x, double y)
        {
            return (Math.Pow((xY - x), 2) + Math.Pow((xY - y), 2) <= radius * radius);
        }

        static bool IsOutsideRectangle(double X, double Y)
        {
            //R(top=1, left=-1, width=6, height=2)

            double left = -1, right = 5, top = 1, bottom = -1;

            return !(Y <= top && Y >= bottom && X >= left && xY <= right);
        }
    }
}

File: Homeworks/JS-basics/Operators-And-Expressions/pointInCircleOutRectangle.js
// Write an expression that checks for given point (x, y)
// if it is within the circle K( (1,1), 3) and out of the
// rectangle R(top=1, left=-1, width=6, height=2).

var rectangle = [1, -1, -1, 5], circle = [1, 1, 3], pointTrue = [2, 2], pointFalse = [1, 1];

console.log(isInsideCircle(pointTrue[0], pointTrue[1], circle[0], circle[1], circle[2])
         && isOutsideRectangle(pointTrue[0], pointTrue[1], rectangle[0], rectangle[1], rectangle[2], rectangle[3]));
console.log(isInsideCircle(pointFalse[0], pointFalse[1], circle[0], circle[1], circle[2])
         && isOutsideRectangle(pointFalse[0], pointFalse[1], rectangle[0], rectangle[1], rectangle[2], rectangle[3]));

function isInsideCircle(x, y, cx, cy, r){
    return (x - cx) * (x - cx) + (y - cy) * (y - cy) < r * r;
}

function isOutsideRectangle(x, y, top, bottom, left, right){
    return !(x >= left && x <= right && y <= top && y >= bottom);
}